sematic/ui/packages/common/src/pages/RunDetails/index.tsx

```diff
@@ -29,16 +38,49 @@ const RunDetails = (props: RunDetailsProps) => {
         return <NotesPane />;
     }, []);
 
-    const contextValue = useMemo(() => {
-        return {
-            rootRun,
-            resolution
-        };
-    }, [rootRun, resolution]);
+    const selectedRun = useMemo(() => {
+        let runId = selectedRunIdHash;
+        if (!runId) {
+            runId = rootRun.id;
+        }
+        if (!graph) {
+            return undefined;
+        }
+        return graph.runsById.get(runId) || rootRun;
+    }, [selectedRunIdHash, graph, rootRun]);
+
+    const rootRunContextValue = useMemo(() => ({
+        rootRun,
+        resolution,
+        graph,
+        isResolutionLoading,
+        isGraphLoading
```

Comment:
Add typing to these arguments?

Reply:
Actually, even if no typings are added, the types will be matched against this line:

```jsx
    return <RootRunContext.Provider value={rootRunContextValue}>
```

So if the `rootRunContextValue` is in bad shape that doesn't match the context type of RootRunContext, the compiler will complain.

I still added typing, FWIW.